File: app/src/main/java/net/ducksmanager/persistence/dao/UserMessageDao.java
package net.ducksmanager.persistence.dao;

import net.ducksmanager.persistence.models.composite.UserMessage;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface UserMessageDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(UserMessage userMessage);

    @Query("SELECT * FROM user_messages WHERE messageKey = :key")
    UserMessage findByKey(String key);

    @Query("DELETE FROM user_messages")
    void deleteAll();
}

File: app/src/main/java/net/ducksmanager/persistence/dao/UserMessageDao.kt
package net.ducksmanager.persistence.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import net.ducksmanager.persistence.models.composite.UserMessage

@Dao
interface UserMessageDao {
    @Query("SELECT * FROM user_messages WHERE messageKey = :key")
    fun findByKey(key: String): UserMessage?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(userMessage: UserMessage)

    @Query("DELETE FROM user_messages")
    fun deleteAll()
}